mlflow/genai/experimental/databricks_trace_exporter.py

```diff
@@ -0,0 +1,387 @@
+
+        Returns:
+            DatabricksTraceDeltaStorageConfig if archival is enabled, None otherwise
+        """
+        with self._config_cache_lock:
+            if experiment_id not in self._config_cache:
+                config = DatabricksTraceServerClient().get_trace_destination(experiment_id)
+                self._config_cache[experiment_id] = config
+                _logger.debug(f"Cached config for experiment {experiment_id}: {config is not None}")
+            return self._config_cache[experiment_id]
+
+    def _archive_trace(self, trace: Trace, experiment_id: str, spans_table_name: str):
+        """
+        Handles exporting a trace to Databricks Delta using the IngestApi.
+
+        Args:
+            trace: MLflow Trace object containing spans data.
+            experiment_id: ID of the experiment for logging.
+            spans_table_name: Name of the spans table for logging.
+        """
+        try:
+            # Convert MLflow trace to OTel proto spans
+            proto_spans = self._convert_trace_to_proto_spans(trace)
+
+            if not proto_spans or len(proto_spans) == 0:
+                _logger.debug("No proto spans to export")
+                return
+
+            # Get stream factory singleton for this table
+            factory = IngestStreamFactory.get_instance(self._spans_table_properties)
+
+            # Get stream from factory
+            stream = factory.get_or_create_stream()
+
+            # Ingest all spans for the trace
+            _logger.debug(
+                f"Ingesting {len(proto_spans)} spans for trace {trace.info.request_id} to table "
+                f"{spans_table_name}"
+            )
+            for proto_span in proto_spans:
+                stream.ingest_record(proto_span)
+
+            # Always flush() to ensure data durability.  In sync logging mode, this is ok.
+            # For async mode, the flush will be handled by the async queue
+            # which runs in the background.
+            stream.flush()
+
+        except Exception as e:
+            _logger.warning(f"Failed to send trace to Databricks Delta: {e}")
+
+    def _convert_trace_to_proto_spans(self, trace: Trace) -> "list[DeltaProtoSpan]":
+        """
+        Convert an MLflow trace to a list of Delta Proto Spans.
+        Direct conversion from MLflow Span to Delta format.
+
+        Args:
+            trace: MLflow Trace object containing spans data.
+
+        TODO: move this to Span.to_proto() once the legacy trace server span are fully repcated
+
+        Returns:
+            List of DeltaProtoSpan objects.
+        """
+        delta_proto_spans = []
+
+        for span in trace.data.spans:
+            delta_proto = DeltaProtoSpan()
+
+            # Use raw OpenTelemetry trace ID instead of the one from the trace
+            # (without "tr-" prefix) for full OTel compliance
+            delta_proto.trace_id = span._trace_id
+            delta_proto.span_id = span.span_id or str(uuid.uuid4())
```

Comment:
```suggestion
            delta_proto.span_id = span.span_id or str(uuid.uuid4())
```

Q: Curious when do we need the `str(uuid.uuid4())` fallback? We should probably discard a span without an ID, because generating random span ID will cause broken parent/child relationship.